ui/src/pages/Create.js

```diff
@@ -0,0 +1,645 @@
+          onClick={(e) => this.setState({showAllIdInfoFields: true})}
+        >
+          <FormattedHTMLMessage {...messages.moreDropdown} />
+        </span>
+      </div>
+    )
+  }
+
+  renderPhotoFields() {
+    var photoFilenameSpan = this.state.formPhotoFile
+        ? <span>{this.state.formPhotoFile.name}</span>
+        : null;
+    var photoDeleteButton = null;
+    if (this.state.formPhotoFile) {
+      photoDeleteButton = (
+        <span onClick={(e) => {this.setState({formPhotoFile: ''})}}>X</span>
+      );
+    }
+    return (
+      <PFNotchedOutline label={this.props.intl.formatMessage(messages.photo)}>
+        <input
+          type='file'
+          ref={this.photoUploadInput}
+          onChange={(e) => this.setState({formPhotoFile: e.target.files[0]})} />
+        <Button
+          className='pf-button-primary'
+          type='button'
+          onClick={() => this.photoUploadInput.current.click()}
+          disabled={this.state.formPhotoUrl != ""}>
+          {this.props.intl.formatMessage(messages.uploadPhoto)}
+        </Button>
+        {photoFilenameSpan}
+        {photoDeleteButton}
+        <TextField
+          label={this.props.intl.formatMessage(messages.orEnterPhotoUrl)}
+          outlined
+        >
+          <Input
+            value={this.state.formPhotoUrl}
+            onChange={(e) => this.setState({formPhotoUrl: e.target.value})}
+            disabled={this.state.formPhotoFile != ""} />
+        </TextField>
+      </PFNotchedOutline>
+    );
+  }
+
+  /*
+   * Adds a profile page field for the given site to the form.
+   */
+  addProfilePageField(site) {
+    this.setState({
+      formProfilePages: this.state.formProfilePages.concat(new Map(
+          [['site', site], ['value', '']])),
+      showProfilePageOptions: false,
+    });
+  }
+
+  /*
+   * Removes the given profile page field from the form.
+   */
+  removeProfilePageField(index) {
+    var newProfileList = this.state.formProfilePages.slice(0, index);
+    if (index < this.state.formProfilePages.length - 1) {
+      newProfileList = newProfileList.concat(
+          this.state.formProfilePages.slice(
+              index + 1, this.state.formProfilePages.length));
+    }
+    this.setState({
+      formProfilePages: newProfileList,
+    });
+  }
+
+  /*
+   * Renders a profile field menu option for the given site.
+   */
+  renderProfilePageOption(site) {
+    // TODO(nworden): include site icons
+    return (
+      <li
+        className='mdc-typography--body1'
+        onClick={() => this.addProfilePageField(site)}>
+        <FormattedMessage {...CREATE_PROFILE_PAGE_SITES[site]} />
+      </li>
+    );
+  }
+
+  /*
+   * Render the profile page link fields (if any) and the "Add site" button.
+   */
+  renderProfilePageFields() {
+    var profilePageFields = this.state.formProfilePages.map((page, index) => (
+      <div
+        className='create-profilefieldwrap'
+        key={this.profileFieldIdCounter++}>
```

Comment:
How about just using `index` as `key`?

Reply:
I considered that at first, and it did seem to work in practice, but the React docs call it a last resort and warn that it'll break things if items can change order (we don't change the order of these, but it makes me nervous about using index if we just happen to fit into a niche case where it works):
https://reactjs.org/docs/lists-and-keys.html#keys
I also came across a couple articles warning against it as an anti-pattern:
https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318
https://menubar.io/react-keys-index
so it just seemed safer to do it this way.

Reply:
It is true that use of an index is not recommended. But I believe always using a new ID like this is even worse.

See https://reactjs.org/docs/reconciliation.html#recursing-on-children for details how keys are used. Couple of things to note:
* (In my understanding) The key only affects performance. So whatever key you give, it won't break the app. It just makes it less efficient.
* React tries to reuse DOM elements with the same key when it re-renders the page. So it helps performance if you give the same key for a component whose content is the same or similar.
* So drawback of using an index as a key is that it cannot reuse the existing DOM elements efficiently when the order is swapped or a component is inserted or removed. But it isn't a big deal for small components like this case.
* If you always give a new ID as a key, the situation gets worse, that it can never reuse existing DOM elements.

Reply:
Ok, I changed it to index with a comment warning that we need to be careful about it.